### PR TITLE
New Destination - Apollo.io 

### PR DESCRIPTION
**Pull request by joe-ayoub-segment**
New Cloud Mode Destination. 
Original [PR raised here ](https://github.com/segmentio/action-destinations/pull/1638) - with review and feedback. Raised this new PR to fix some tests. 

## Testing

tested locally

### CODE REVIEW

File: packages/destination-actions/src/destinations/apolloio/track/index.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+          description: 'The search query of the page where the event occurred.',
+          label: 'Search'
+        }
+      }
+    }
+  },
+  perform: (request, data) => {
+    return request(baseURL, {
+      method: 'post',
+      json: [data.payload]
```
Comment (varadarajan-tw) [inline packages/destination-actions/src/destinations/apolloio/track/index.ts:153]: Why is the payload array here but single event in `performBatch`?  Maybe the two needs to be swapped?

File: packages/destination-actions/src/destinations/apolloio/track/index.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+      default: { '@path': '$.event' }
+    },
+    properties: {
+      type: 'object',
+      label: 'Properties',
+      description: 'Properties to associate with the event',
+      default: { '@path': '$.properties' }
+    },
+    timestamp: {
+      type: 'string',
```
Comment (varadarajan-tw) [inline packages/destination-actions/src/destinations/apolloio/track/index.ts:31]: Why not use datetime?

File: packages/destination-actions/src/destinations/apolloio/index.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import type { Settings } from './generated-types'
+import { IntegrationError, defaultValues } from '@segment/actions-core'
+
+import track from './track'
+export const baseURL = process?.env?.ACTIONS_APOLLOIO_BASE_URL_SECRET
+  ? `${process.env.ACTIONS_APOLLOIO_BASE_URL_SECRET}`
+  : 'https://apollo.io/'
+export const authURL = process?.env?.ACTIONS_APOLLOIO_AUTH_URL_SECRET
+  ? `${process.env.ACTIONS_APOLLOIO_AUTH_URL_SECRET}`
+  : 'https://apollo.io/'
```
Comment (varadarajan-tw) [inline packages/destination-actions/src/destinations/apolloio/index.ts:11]: nit: This is a lot more succinct. [Null coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing)
```suggestion
export const authURL = process?.env?.ACTIONS_APOLLOIO_AUTH_URL_SECRET ?? 'https://apollo.io/'
```

**Review by varadarajan-tw — APPROVED** [review]
LGTM. Left few comments.

File: packages/destination-actions/src/destinations/apolloio/track/index.ts (hunk 1 above)
Comment (joe-ayoub-segment) [reply packages/destination-actions/src/destinations/apolloio/track/index.ts:153]: I think it's already correct - the Partner wants to send an array regardless. And data.payload is already an array in the performBatch function.

**Comment by joe-ayoub-segment** [comment]
note: PR deployed

**Pull request merged**